### PR TITLE
Select artifacts from all available variants

### PR DESCRIPTION
**Pull request by big-guy**

### CODE REVIEW

**Comment by big-guy** [comment]
This is blocked on #19895

**Comment by big-guy** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
@bot-gradle test this

File: subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/MetadataSourcedComponentArtifacts.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
     @Override
     public ArtifactSet getArtifactsFor(ComponentResolveMetadata component, ConfigurationMetadata configuration, ArtifactResolver artifactResolver, Map<ComponentArtifactIdentifier, ResolvableArtifact> allResolvedArtifacts, ArtifactTypeRegistry artifactTypeRegistry, ExcludeSpec exclusions, ImmutableAttributes overriddenAttributes, CalculatedValueContainerFactory calculatedValueContainerFactory) {
         Set<VariantResolveMetadata> variants = getVariantResolveMetadata(component, configuration);
-        return DefaultArtifactSet.createFromVariantMetadata(component.getId(), component.getModuleVersionId(), component.getSources(), exclusions, variants, component.getAttributesSchema(), artifactResolver, allResolvedArtifacts, artifactTypeRegistry, overriddenAttributes, calculatedValueContainerFactory);
+        return DefaultArtifactSet.createFromVariantMetadata(component.getId(), component.getModuleVersionId(), component.getSources(), exclusions, variants, configuration.getVariants(), component.getAttributesSchema(), artifactResolver, allResolvedArtifacts, artifactTypeRegistry, overriddenAttributes, calculatedValueContainerFactory);
```
Comment (DPUkyle) [inline subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/MetadataSourcedComponentArtifacts.java:45]: How are the two method parameters `variants` and `configuration.getVariants()` going to differ IRL?

**Comment by big-guy** [comment]
@bot-gradle help

**Comment by big-guy** [comment]
@bot-gradle test RFR

**Comment by big-guy** [comment]
@bot-gradle test RFR

**Comment by big-guy** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
@bot-gradle test RFR

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Pull request merged**